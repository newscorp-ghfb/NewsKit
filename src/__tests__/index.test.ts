import * as newskit from '..';

describe('index exports', () => {
  test('all expected functions and components are exported', () => {
    expect(Object.keys(newskit).sort()).toMatchSnapshot();
  });

  test('no styled components are exported', () => {
    expect(
      // We should be prefixing all our internal styled comps with Styled - and these should not be exported.
      Object.keys(newskit)
        .sort()
        .filter(exportKey => exportKey.startsWith('Styled')),
    ).toEqual([]);
  });
});
