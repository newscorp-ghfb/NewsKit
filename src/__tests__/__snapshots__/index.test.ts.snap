// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index exports all expected functions and components are exported 1`] = `
Array [
  "A",
  "AlignSelfValues",
  "AspectRatio",
  "AssistiveText",
  "AudioEvents",
  "AudioPlayer",
  "Banner",
  "Block",
  "Button",
  "ButtonSize",
  "ButtonsContainer",
  "Byline",
  "Caption",
  "CaptionInset",
  "Card",
  "CardInset",
  "Cell",
  "Checkbox",
  "Consent",
  "ConsentSettingsLink",
  "ControlPanel",
  "DateTime",
  "Devices",
  "Divider",
  "Drawer",
  "EmailInput",
  "EventTrigger",
  "ExperimentationWeb",
  "Fieldset",
  "Flag",
  "FlagSize",
  "Flow",
  "Form",
  "FormController",
  "FormInput",
  "FormInputAssistiveText",
  "FormInputCheckbox",
  "FormInputLabel",
  "FormInputSelect",
  "FormInputTextField",
  "Global",
  "Grid",
  "H1",
  "H2",
  "H3",
  "H4",
  "H5",
  "H6",
  "Heading1",
  "Heading2",
  "Heading3",
  "Heading4",
  "Heading5",
  "Heading6",
  "Headline",
  "Hidden",
  "IconButton",
  "Image",
  "IndeterminateProgressIndicator",
  "InlineDrawer",
  "InlineMessage",
  "InstrumentationProvider",
  "Label",
  "LabelPosition",
  "Link",
  "LinkInline",
  "LinkStandalone",
  "MediaQueryProvider",
  "Menu",
  "MenuDivider",
  "MenuGroup",
  "MenuItem",
  "Modal",
  "OrderedList",
  "P",
  "Paragraph",
  "ParagraphDropCap",
  "ParagraphText",
  "PopoutButton",
  "ScreenReaderOnly",
  "Scroll",
  "ScrollSnapAlignment",
  "Select",
  "SelectOption",
  "ShareBar",
  "Slider",
  "Stack",
  "StackChild",
  "StackDistribution",
  "Standfirst",
  "StatefulSlider",
  "StructuredList",
  "StructuredListCell",
  "StructuredListItem",
  "Sub",
  "Subscript",
  "Sup",
  "Superscript",
  "Svg",
  "Tab",
  "TabAlign",
  "TabSize",
  "Tabs",
  "TabsDistribution",
  "TabsIndicatorPosition",
  "Tag",
  "TagSize",
  "Tealium",
  "TextBlock",
  "TextField",
  "TextFieldSize",
  "TextInput",
  "TextInputSize",
  "ThemeProvider",
  "ThemelessAssistiveText",
  "ThemelessByline",
  "ThemelessCaption",
  "ThemelessLabel",
  "ThemelessTextField",
  "TitleBar",
  "Toast",
  "ToastProvider",
  "UnorderedList",
  "VideoPlayer",
  "Visible",
  "VolumeControl",
  "compileTheme",
  "composeInstrumentationMiddleware",
  "createEventInstrumentation",
  "createTheme",
  "css",
  "customToNewsKitIcon",
  "deepMerge",
  "getAspectRatioStyles",
  "getBorderCssFromTheme",
  "getBorderFromTheme",
  "getColorCssFromTheme",
  "getColorFromTheme",
  "getDefaultedValue",
  "getDeviceQueryFromTheme",
  "getFontSizingFromTheme",
  "getFontsFromTheme",
  "getHeight",
  "getMarginPresetFromTheme",
  "getMaxSize",
  "getMediaQueries",
  "getMediaQuery",
  "getMediaQueryFromTheme",
  "getMinHeight",
  "getMinSize",
  "getMinWidth",
  "getMotionCssFromTheme",
  "getMotionDurationValue",
  "getMotionFromTheme",
  "getOverlayCssFromTheme",
  "getPresetStyles",
  "getResponsiveBorder",
  "getResponsiveMotion",
  "getResponsiveSize",
  "getResponsiveSpace",
  "getResponsiveSpacingInlineHorizontal",
  "getResponsiveSpacingInlineVertical",
  "getResponsiveSpacingInset",
  "getResponsiveSpacingStackHorizontal",
  "getResponsiveSpacingStackVertical",
  "getResponsiveValueFromTheme",
  "getResponsiveX",
  "getSSRId",
  "getShadowCssFromTheme",
  "getShadowFromTheme",
  "getSingleStylePreset",
  "getSize",
  "getSizing",
  "getSizingCssFromTheme",
  "getSizingFromTheme",
  "getSpace",
  "getSpacingCssFromTheme",
  "getSpacingFromTheme",
  "getSpacingInlineHorizontal",
  "getSpacingInlineVertical",
  "getSpacingInset",
  "getSpacingInsetFromTheme",
  "getSpacingStackHorizontal",
  "getSpacingStackVertical",
  "getStylePreset",
  "getStylePresetFromTheme",
  "getTransitionDuration",
  "getTransitionDurationFromTheme",
  "getTypographyPreset",
  "getTypographyPresetFromTheme",
  "getValueFromTheme",
  "getWeight",
  "getWidth",
  "getXFromTheme",
  "handleResponsiveProp",
  "instrumentationHandlers",
  "instrumentationMiddleware",
  "isButtonLink",
  "isResponsive",
  "keyframes",
  "mergeContexts",
  "newskitDarkTheme",
  "newskitDarkThemeOverrides",
  "newskitLightTheme",
  "styled",
  "toNewsKitIcon",
  "toast",
  "useBreakpointKey",
  "useClientSide",
  "useFormContext",
  "useInstrumentation",
  "useMediaQueryObject",
  "useTheme",
  "withDefaultProps",
  "withInstrumentation",
  "withTheme",
]
`;
